commands/apps_dev.go

```diff
@@ -241,29 +235,49 @@ func RunAppsDevBuild(c *CmdConfig) error {
 		}
 	}
 
+	var (
+		// componentConfig is the per-component config source.
+		componentConfig = appsDevFlagConfigCompat(conf.Components(component))
+		// componentGlobalConfig is per-component config that can be overridden at the global/app level as well as via cli flags.
+		componentGlobalConfig = config.Multi(
+			doctlConfig,
+			componentConfig,
+			appsDevFlagConfigCompat(conf),
+		)
+		// componentArgConfig is per-component config that can be overridden via cli flags.
+		componentArgConfig = config.Multi(
```

Comment:
why do we need this? Shouldn't this be covered by the `componentGlobalConfig` since first match would win?

Reply:
`componentGlobalConfig` would allow for an app-wide `build_command` in `dev-config.yaml` which wouldn't make much sense

Reply:
Would it really allow for it? We enforce setting through the `app dev config set` so it would just simplify the accessor pattern in my mind.